mdm/mdm/command.go

```diff
@@ -16,8 +17,16 @@ type CommandPayload struct {
 }
 
 func NewCommandPayload(request *CommandRequest) (*CommandPayload, error) {
+	var CommandUUID string
+	// The `CommandUUID` parameter is optional. If it was not sent, the default behavior is to provide one randomly.
+	if request.CommandUUID != nil {
+		CommandUUID = *request.CommandUUID
+	} else {
+		CommandUUID = uuid.New().String()
+	}
+
 	payload := &CommandPayload{
-		CommandUUID: uuid.New().String(),
+		CommandUUID: CommandUUID,
```

Comment:
To make this simpler/cleaner maybe: Unconditionally set the request/incoming UUID, then test for emptiness and set a UUID before returning `payload` in a single `if`, avoiding an `else`.

Reply:
I was going to make the same suggestion! Also lowercase variable names please :) `commandUUID`